lib/active_record/connection_adapters/sqlserver/schema_creation.rb

```diff
@@ -10,8 +10,10 @@ def visit_TableDefinition(o)
             table_name = quote_table_name(o.temporary ? "##{o.name}" : o.name)
             query = o.as.respond_to?(:to_sql) ? o.as.to_sql : o.as
             projections, source = query.match(%r{SELECT\s+(.*)?\s+FROM\s+(.*)?}).captures
-            select_into = "SELECT #{projections} INTO #{table_name} FROM #{source}"
+            "SELECT #{projections} INTO #{table_name} FROM #{source}"
           else
+            raise NotImplementedError.new("SQL Server does not support 'CREATE TABLE IF NOT EXISTS [table_name]'") if o.if_not_exists
```

Comment:
We should try to be compliant with ActiveRecord public API and implement the `if_exists` if possible. I know this is going to require a lot of changes but if it is possible we should try to keep the adapter compatible with all other databases.